### ISSUE TITLE
Pipeworks prevents uncaughtException event to be fired

### ISSUE THREAD

**Issue by AtaS** [kevinswiber/pipeworks#8]
Hi,

Inside pipeworks.js there is a code `this.domain.on('error'...`

This code prevents `process.on('uncaughtException', callback)` to be fired (*in some places only*) when there is an uncaught exception. 

That event is very important for logging the exception that was not caught before the node process exists.

How can we overcome this problem, do you have a suggestion?

Thanks

**Comment by AtaS** [comment]
@kevinswiber  Any ideas on this?
I think Swagger Express module is using your library through Bagpipes.

**Comment by kevinswiber** [comment]
I'll take a look.

**Comment by kevinswiber** [comment]
Do you have an example of what you're trying to do that demonstrates the
problem?

**Comment by AtaS** [comment]
Okay the workflow is something like this:

When there is an uncaught exception/error in the project my handler gets fired only if that is not through an HTTP request.

When there is an HTTP request, some files/services are 'require'd by Swagger and I am kinda in "Swagger" scope, which decides which method to execute by the given URL mapping. When I run the piece of code below through one of the libraries that is executed when an HTTP request happens then my "uncaughtException" handler doesn't get fired.

```js
setTimeout(function() {
     throw new Error('MY ERROR');
}, 500);
```

Instead `this.domain.on('error'...)` gets fired in pipeworks.js without `self.faultPipe` so basically it just `rethrow`s the error but then my uncaughtException handler still doesn't get fired.

So I am losing information about why my node process is crashing because I can't log it when this happens.

**Comment by AtaS** [comment]
And this is the only thing I get without my uncaughtException even was fired:

```
/Users/asasmaz/.../node_modules/pipeworks/pipeworks.js:204
      throw err; // rethrow
      ^
Error: MY ERROR
    at null._onTimeout (/Users/asasmaz/.../myJS.js:74:27)
    at Timer.listOnTimeout (timers.js:92:15)
```

**Comment by AtaS** [comment]
Hi @kevinswiber ,
This is perfect thanks :)
Now my custom uncaughtException event is getting fired where it wasn't getting fired.
Cheers!
Ata

**Comment by kevinswiber** [comment]
The fix has been published in `pipeworks@1.3.1`.  Thanks!
